pkg/metrics/machinehealthcheck.go

```diff
@@ -0,0 +1,47 @@
+/*
+   Copyright 2020 The Machine API Operator authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	DefaultHealthCheckMetricsAddress = ":8083"
+)
+
+var (
+	// MachineHealthCheckNodesCovered is a Prometheus metric, which reports the number of nodes covered by MachineHealthChecks
+	MachineHealthCheckNodesCovered = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "mapi_machinehealthcheck_nodes_covered",
+			Help: "Number of nodes covered by MachineHealthChecks",
+		}, []string{"name"},
+	)
+)
+
+func InitializeMachineHealthCheckMetrics() {
+	metrics.Registry.MustRegister(MachineHealthCheckNodesCovered)
+}
+
+func DeleteMachineHealthCheckNodesCovered(name string) {
+	MachineHealthCheckNodesCovered.Delete(prometheus.Labels{"name": name})
+}
+
+func RegisterMachineHealthCheckNodesCovered(name string, count int) {
```

Comment:
Register feels like a weird name here since it is used in the `registry.MustRegister` context as well, would `Update` or `Observe` be better?

Reply:
i tried to follow the naming convention for the other metrics, but i can certainly make these more specific.